### PR TITLE
Added new selectable gradient + accent colors

### PR DESCRIPTION
**Pull request by marcelweikum**
Hey Victor,

I now added the feature which I requested earlier. https://github.com/victorbalssa/abacus/issues/319#issue-2413149510

Main changes are:

- New Color Selection Screen in the Configuration Screen:
  - Here you can select a color that is saved to redux storage. Even after a restart the selected color is still selected.

- The gradients are now changed into a compacted version, so you only have to declare them in colors.ts

- I added translations for all available languages for my feature

- Gradients are linked to Accent Color, when I change to a color, the gradient on the Home Screen as well as the colors for the Buttons, Graphs (only first graph, the others are random) and Switches change too.

- I changed the TransactionForm Input Background to a more transparent version of the selected Color (That technically is the same as before but now with different colors)

Simulator Screenshot - iPhone 15 - 2024-07-17 at 21 46 45

Simulator Screenshot - iPhone 15 - 2024-07-17 at 21 46 28

Simulator Screenshot - iPhone 15 - 2024-07-17 at 21 47 08

If there are any suggestions or bugs, do not hesitate to contact me!

Best regards,
Marcel

### CODE REVIEW

File: src/components/Screens/HomeScreen.tsx [hunk 1]
```diff
@@ -538,6 +539,11 @@ export default function HomeScreen() {
   const end = useSelector((state: RootState) => state.firefly.rangeDetails.end);
   const currentCode = useSelector((state: RootState) => state.currencies.currentCode);
   const dispatch = useDispatch<RootDispatch>();
+  const selectedTheme = useSelector((state: RootState) => state.configuration.selectedTheme);
```
Comment (victorbalssa) [inline src/components/Screens/HomeScreen.tsx]: When user update the app via OTA new store key is `undefined` so we need a way to default to `gradientOrange` until the user select another theme:
 ```suggestion
  const selectedTheme = useSelector((state: RootState) => state.configuration.selectedTheme || 'gradientOrange');```

File: src/components/UI/ALibrary/AButton.tsx [hunk 2]
```diff
@@ -29,7 +31,8 @@ export default function AButton({
   testID = null,
 }: AButtonType) {
   const { colors } = useThemeColors();
-  const backgroundColor = colors[type];
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/components/UI/ALibrary/AButton.tsx]: same comment here we need to default to `colors.brandStyleOrange`

File: src/components/UI/ALibrary/AInput.tsx [hunk 3]
```diff
@@ -52,19 +54,28 @@ export default function AInput({
 }: AInputType) {
   const { colors } = useThemeColors();
   const [isFocused, setIsFocused] = React.useState(false);
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/components/UI/ALibrary/AInput.tsx]: same comment here we need to default to `colors.brandStyleOrange`

File: src/components/UI/Filters.tsx [hunk 4]
```diff
@@ -22,6 +22,7 @@ export default function Filters() {
   const range = useSelector((state: RootState) => state.firefly.rangeDetails.range);
   const accounts = useSelector((state: RootState) => state.accounts.accounts);
   const selectedAccountIds = useSelector((state: RootState) => state.accounts.selectedAccountIds);
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/components/UI/Filters.tsx]: same comment here we need to default to `colors.brandStyleOrange`

File: src/components/Screens/ConfigurationScreen.tsx [hunk 5]
```diff
@@ -37,6 +37,7 @@ export default function ConfigurationScreen({ navigation }: ScreenType) {
   const backendURL = useSelector((state: RootState) => state.configuration.backendURL);
   const useBiometricAuth = useSelector((state: RootState) => state.configuration.useBiometricAuth);
   const dispatch = useDispatch<RootDispatch>();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/components/Screens/ConfigurationScreen.tsx]: same comment here we need to default to `colors.brandStyleOrange`

File: src/components/Screens/ColorSelectionScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  },
+  selectedTile: {
+    borderWidth: 2,
+    borderColor: 'black',
+  },
+});
+
+export default function ColorSelectionScreen() {
+  const { colors: themeColors } = useThemeColors();
+  const currentTheme = useSelector((state: RootState) => state.configuration.selectedTheme);
```
Comment (victorbalssa) [inline src/components/Screens/ColorSelectionScreen.tsx]: same comment here we need to default to `gradientOrange`

File: src/components/Screens/FilterScreen.tsx [hunk 7]
```diff
@@ -10,8 +10,10 @@ import { useThemeColors } from '../../lib/common';
 import { ScreenType } from '../../types/screen';
 import { types } from '../../models/transactions';
 import translate from '../../i18n/locale';
+
 export default function FilterScreen({ navigation, route }: ScreenType) {
   const { colors } = useThemeColors();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/components/Screens/FilterScreen.tsx]:
```suggestion
  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle || colors.brandStyleOrange);
```

File: src/components/Screens/HomeScreen.tsx [hunk 8]
```diff
@@ -44,6 +44,7 @@ function AssetsAccounts() {
   const displayAllAccounts = useSelector((state: RootState) => state.configuration.displayAllAccounts);
   const loading = useSelector((state: RootState) => state.loading.effects.accounts.getAccounts?.loading);
   const dispatch = useDispatch<RootDispatch>();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/components/Screens/HomeScreen.tsx]:
```suggestion
  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle || colors.brandStyleOrange);
```

File: src/routes/index.tsx [hunk 9]
```diff
@@ -77,12 +79,13 @@ const styles = StyleSheet.create({
 function TabBarPrimaryButton() {
   const navigation = useNavigation();
   const { colors } = useThemeColors();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
Comment (victorbalssa) [inline src/routes/index.tsx]:
```suggestion
  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle || colors.brandStyleOrange);
```

**Review by victorbalssa — COMMENTED** [review]
Thanks @marcelweikum, review comments follow.

File: src/components/Screens/HomeScreen.tsx (hunk 1 above)
Comment (marcelweikum) [reply src/components/Screens/HomeScreen.tsx]: True! Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/UI/ALibrary/AButton.tsx (hunk 2 above)
Comment (marcelweikum) [reply src/components/UI/ALibrary/AButton.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/UI/ALibrary/AInput.tsx (hunk 3 above)
Comment (marcelweikum) [reply src/components/UI/ALibrary/AInput.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/UI/Filters.tsx (hunk 4 above)
Comment (marcelweikum) [reply src/components/UI/Filters.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/Screens/ConfigurationScreen.tsx (hunk 5 above)
Comment (marcelweikum) [reply src/components/Screens/ConfigurationScreen.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/Screens/ColorSelectionScreen.tsx (hunk 6 above)
Comment (marcelweikum) [reply src/components/Screens/ColorSelectionScreen.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/Screens/FilterScreen.tsx (hunk 7 above)
Comment (marcelweikum) [reply src/components/Screens/FilterScreen.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/components/Screens/HomeScreen.tsx (hunk 8 above)
Comment (marcelweikum) [reply src/components/Screens/HomeScreen.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

File: src/routes/index.tsx (hunk 9 above)
Comment (marcelweikum) [reply src/routes/index.tsx]: Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

**Comment by marcelweikum** [comment]
I've now added the initial color/gradient for the missing init key.

Also I've changed the ColorSelection to now fit more to the surrounding colors and removed the ScrollView.

Simulator Screenshot Dark
Simulator Screenshot Light

Another fix I made is the highlighting of the selected color, this was black all the time, no matter if dark mode or light mode is set. Now it sets depending on the mode.

**Comment by marcelweikum** [comment]
Hey again,

I found a missing color Adjustment in efa9c2106c85ad5891519978dbf2dc278c4f7547 and fixed it.

**Review by victorbalssa — APPROVED** [review]
Thank you @marcelweikum, approved.

I will include this work in the next release.

**Pull request merged**